src/EnergyPlus/GeneratorDynamicsManager.cc

```diff
@@ -225,30 +225,25 @@ namespace GeneratorDynamicsManager {
         Real64 PelInput;                        // holds initial value of IN var
         DataGenerators::OperatingMode newOpMode(DataGenerators::OperatingMode::Invalid);
         Real64 ElectLoadForThermalRequest;
-        auto &InletCWnode = state.dataGenerator->InletCWnode;
-        auto &InternalFlowControl = state.dataGenerator->InternalFlowControl;
-        auto &TcwIn = state.dataGenerator->TcwIn;
-        auto &TrialMdotcw = state.dataGenerator->TrialMdotcw;
-        auto &LimitMinMdotcw = state.dataGenerator->LimitMinMdotcw;
```

Comment:
@amirroth I have a question here about loading these vars to a register for use in this function. I realize the # of registers is limited but do we not want to do this?

Reply:
This has less to do with registers and more to do with the fact that these variables are written to and so if you use a reference shortcut, it looks like you are writing to a local variable and not that you are changing global state. 

However, since you mentioned it, reference shortcuts to scalars also actively slow you down. 
When you have a reference to X what is actually stored in the register is the address of X. Suppose `TcwIn` is the third member in the `GeneratorData struct`, in other words its address is 16 bytes after the start of the struct. Suppose we have the `state.dataGenerator` in register `R1`. Then `auto &TcwIn = state.dataGenerator->TcwIn;` compiles to `ADD R1, 16 -> R2`. And using the value in `TcwIn` requires you to read that value into another register `LOAD R2 -> R3`. 

Here is the thing, in basically all instruction sets (x64 included), the address calculation that is part of a `LOAD` or `STORE` includes the addition of a constant offset because that is such a common idiom. `LOAD R2 -> R3` is really `LOAD R2+0 -> R3`. You get that one addition essentially "for free" when you the do the `LOAD` or `STORE`. And so breaking that addition away from the `LOAD` or `STORE` using a shortcut reference is not actually getting you anything. In fact, it's making you do two instructions where you could have just as easily done one. If you want to read the value of `TcwIn` with the reference shortcut, you get:

```cpp
ADD R1, 16 -> R2
LOAD R2+0 -> R3
```

And without the shortcut, you get:

```cpp
LOAD R1+16 -> R3
```

And same for `STORE` for writing through the shortcut. Reference shortcuts to scalars are bad on multiple levels.